src/v2/ion/ViewClassNamesFactory.js

```diff
@@ -0,0 +1,88 @@
+/* eslint complexity: [2, 25] */
+import {FORMS} from './RemediationConstants';
+
+const FORMNAME_CLASSNAME_MAPPINGS = {
+  [FORMS.IDENTIFY]: {
+    'identify': ['identify', 'primary-auth'],
+    'password': ['identify', 'primary-auth']
+  },
+  [FORMS.ENROLL_PROFILE]: {
+    'enroll-profile': ['enroll-profile', 'registration'],
+  },
+  [FORMS.CHALLENGE_AUTHENTICATOR]: {
+    email: ['challenge-authenticator-email', 'mfa-verify-passcode'],
```

Comment:
I thought we gonna remove the redundancy like `challenge-authenticator-email` ?

Reply:
What is the redundancy here ? For the `challenge-authenticator form` with `email` authenticator  we are adding a `challenge-authenticator-email` class. 

We could generate this dynamically and only keep the override className here but I think its easier to know what classes are getting added if they are all in the same place.

Reply:
because we suppose to generate `challenge-authenticator-email` instead of hard code. 
At lease `${FORMS.CHALLENGE_AUTHENTICATOR}-email`?

Readability is good point. I feel it's fine as long as we can make clear about rule that how addition css class are generated.

Reply:
yeah I did think about that but I left it as is since part `${FORMS.CHALLENGE_AUTHENTICATOR}-email` is still hardcoded. But I don't feel too strongly about it. I can update it to use the formName for now